application/libraries/Api/Rest/Renderer/RendererBasic.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace LimeSurvey\Api\Rest\Renderer;
+
+use CJSON;
+use CWebLogRoute;
+use Exception;
+use Yii;
+
+use LimeSurvey\Api\Rest\Renderer\RendererInterface;
+use LimeSurvey\Api\Command\Response\{
+    Response,
+    Status
+};
+
+class RendererBasic implements RendererInterface
+{
+    public function returnResponse(Response $response)
+    {
+        $this->renderJSON(
+            $response->getData(),
+            $this->getHttpResponseCode($response->getStatus())
+        );
+    }
+
+    public function returnException(Exception $exception)
+    {
+        $error = [];
+
+        $error['code'] = get_class($exception);
+        if (YII_DEBUG) {
+            $error['message'] = $exception->getMessage();
+            $error['file'] = $exception->getFile();
+            $error['line'] = $exception->getLine();
+            $error['stacktrace'] = $exception->getTraceAsString();
+        }
+
+        $this->renderJson(['error' => $error], 500);
+    }
+
+    /**
+     * Return data to browser as JSON and end application.
+     *
+     * @param array $data
+     * @param int $responseCode
+     * @return void
+     */
+    protected function renderJSON($data, $responseCode = 200)
+    {
+        http_response_code($responseCode);
+        header('Content-type: application/json');
+        echo CJSON::encode($data);
+
+        foreach (Yii::app()->log->routes as $route) {
+            if ($route instanceof CWebLogRoute) {
+                $route->enabled = false; // disable any weblogroutes
+            }
+        }
+        Yii::app()->end();
+    }
+
+    /**
+     * Get HTTP response code from command response status.
+     *
+     * @param StatusAbstract $status
+     * @return void
```

Comment:
NIT: I think it will be more appropriate to make it `@return int The corresponding HTTP response code.`

Reply:
What is the advantage? 

I know it seems nicer / more symmetrical to have methods return something but I don't see any reason to return a value here.